pkg/netpol/eval/internal/k8s/netpol.go

```diff
@@ -29,14 +29,35 @@ import (
 	"github.com/np-guard/netpol-analyzer/pkg/netpol/internal/common"
 )
 
-// NetworkPolicy is an alias for k8s network policy object
-// @todo is there a preprocessed form of the object that would make more sense?
+// @todo is there another preprocessed form of the object that would make more sense?
 //
 //	for example, converting Spec.PodSelector to labels.Selector on initialization
 //	or preprocessing namespaces and pods that match selector in ingress/egress rules, etc
 //
 // -> might help to preprocess and store peers that match policy selectors + selectors in rules + set of allowed connections per rule
-type NetworkPolicy netv1.NetworkPolicy
+type NetworkPolicy struct {
+	*netv1.NetworkPolicy // embedding k8s network policy object
+	// following data stored in preprocessing;
+	// IngressGeneralConns contains:
+	// - the maximal connection-set which the policy's rules allow to all destinations on ingress direction
+	// - the maximal connection-set which the policy's rules allow to all namespaces in the cluster on ingress direction
+	IngressGeneralConns PolicyGeneralRulesConns
+	// EgressGeneralConns contains:
+	// - the maximal connection-set which the policy's rules allow to all destinations on egress direction
+	// - the maximal connection-set which the policy's rules allow to all namespaces in the cluster on egress direction
+	EgressGeneralConns PolicyGeneralRulesConns
+
+	namedPortsToPortNum map[v1.Protocol]map[string]int32 // internal map; map from protocol to map from namedPort which
```

Comment:
the matching of named port to its number is per pod, so why is it stored here?

Reply:
answered in the previous comment

Reply:
fixed the handling of named-ports

**adding some examples - locally tested :**

1. an ingress policy; exposes the workload on ingress to entire-cluster on named ports; since the dst is the workload itself - converts the ports to numbers

![ing_spec_result](https://github.com/np-guard/netpol-analyzer/assets/82180114/7b59ad58-bc89-4850-a7bf-03328dd75d31)
 
2. an egress policy - which exposes the workload to entire cluster ; on named ports : we will see these potential named ports on the result.

![egress_res](https://github.com/np-guard/netpol-analyzer/assets/82180114/5f9e6d40-e666-4046-b73a-91968ffd1256)

3. an example combining named ports and port numbers:

![example_spec_res](https://github.com/np-guard/netpol-analyzer/assets/82180114/f2cb85bf-3830-471b-a085-61e7b16cfb11)

Reply:
thanks, the example seem to make sense